components/script/dom/range.rs

```diff
@@ -658,6 +658,9 @@ impl RangeMethods for Range {
         let start_offset = self.StartOffset();
 
         // Step 1.
+        if &*start_node == node {
+            return Err(Error::HierarchyRequest);
+        }
         match start_node.type_id() {
             // Handled under step 2.
             NodeTypeId::CharacterData(CharacterDataTypeId::Text) => (),
```

Comment:
This probably needs fixing as well, according to the spec.
> If range’s start node is a ProcessingInstruction or Comment node, is a Text node whose parent is null, or is node, throw an HierarchyRequestError exception.

It needs to check whether start's parent is null.